FirebaseDynamicLinks/Sources/FIRDLDefaultRetrievalProcessV2.m

```diff
@@ -296,6 +296,35 @@ - (nullable NSURL *)uniqueMatchLinkToCheck {
   return nil;
 }
 
+- (NSString *)retrievePasteboardContents {
+  NSString *pasteboardContents;
+#if defined(__IPHONE_14_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
+  if (@available(iOS 14.0, *)) {
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    __block NSString *detectedContents;
+    NSSet<UIPasteboardDetectionPattern> *detectionPatterns =
+        [NSSet setWithArray:@[ UIPasteboardDetectionPatternProbableWebURL ]];
+    [[UIPasteboard generalPasteboard]
+        detectPatternsForPatterns:detectionPatterns
+                completionHandler:^(NSSet<UIPasteboardDetectionPattern> *patternSet,
+                                    NSError *error) {
+                  if ([patternSet containsObject:UIPasteboardDetectionPatternProbableWebURL]) {
+                    detectedContents = [UIPasteboard generalPasteboard].string;
+                  }
+                  dispatch_semaphore_signal(semaphore);
+                }];
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
```

Comment:
wdyt about using a completion handler for retrievePasteboardContents instead of waiting and returning a string from this method?

Reply:
I think that is a valid approach too but Im not sure how that would change initial call [here](https://github.com/firebase/firebase-ios-sdk/pull/5905/files#diff-b0b41bd29c92c63bfcb04da66a305f35L273). It might require more refactoring for that method, which im not sure if that is the right move considering this proposal PR is not really a fix. Im not too sharp at handling async methods in objc c though so if you can make it cleaner, feel free! 🙂

Reply:
For proposal PR it should be fine. Otherwise as you can see, this method will get called from app start up and we don't want to freeze the app by waiting here to check the pasteboard.